service/src/main/java/tds/config/services/impl/TimeLimitConfigurationServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package tds.config.services.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import tds.config.TimeLimitConfiguration;
+import tds.config.repositories.TimeLimitConfigurationRepository;
+import tds.config.services.TimeLimitConfigurationService;
+
+import java.util.Optional;
+
+@Service
+public class TimeLimitConfigurationServiceImpl implements TimeLimitConfigurationService {
+    private final TimeLimitConfigurationRepository timeLimitConfigurationRepository;
+
+    @Autowired
+    public TimeLimitConfigurationServiceImpl(TimeLimitConfigurationRepository timeLimitConfigurationRepository) {
+        this.timeLimitConfigurationRepository = timeLimitConfigurationRepository;
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName);
+    }
+
+    @Override
+    public Optional<TimeLimitConfiguration> findTimeLimitConfiguration(String clientName, String assessmentId) {
+        return timeLimitConfigurationRepository.findTimeLimitConfiguration(clientName, assessmentId)
+                .map(Optional::of)
```

Comment:
Actually, it is - if I remove the `.map(Optional::of)` line, the code won't compile.  I'm still not 100 sure why; it must have something to do with handling the `orElse()` call.  It's on my list of things to look into.  But I can say if I don't call `map(Optional::of)` before calling `orElse()`, it no worky.